### PR TITLE
[Form] add test for ArrayChoiceList handling null

### PR DESCRIPTION
**Pull request by Tobion**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I found this bug but then realised that it was already fixed in #17511.
So this just adds a test.

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/ChoiceList/ArrayChoiceListTest.php [hunk 1]
```diff
@@ -18,13 +18,9 @@
  */
 class ArrayChoiceListTest extends AbstractChoiceListTest
 {
-    private $object;
-
     protected function setUp()
     {
         parent::setUp();
-
-        $this->object = new \stdClass();
```
Comment (Tobion) [inline src/Symfony/Component/Form/Tests/ChoiceList/ArrayChoiceListTest.php]: this was broken because it was initialized after `getChoices` is called. so `$this->object` was actually null.

**Comment by fabpot** [comment]
Looks like this breaks the tests

**Comment by xabbuh** [comment]
Using `assertSame()` now doesn't work as the object isn't the same instance anymore.

File: src/Symfony/Component/Form/Tests/ChoiceList/ArrayChoiceListTest.php (hunk 1 above)
Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/ChoiceList/ArrayChoiceListTest.php]: You should maybe just put it before the call to `parent::setUp()` instead of removing it and keep using `$this->object` in `getChoices()`

**Comment by Tobion** [comment]
Fixed

**Comment by HeahDude** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @Tobion.
